src/convertToObject.js

```diff
@@ -6,7 +6,17 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const rules = sourceString.split(';');
+
+  const result = {};
+
+  rules.forEach((rule) => {
+    const [key, value] = rule.split(':').map((part) => part.trim());
+
+    result[key] = value;
```

Comment:
[CODE PRACTICE] - The current implementation does not handle the case where there might be empty keys or values as a result of splitting by semicolon. It's important to filter out such cases to ensure the object does not contain undefined or empty keys. You can achieve this by checking if the key is truthy before assigning it to the result object.